account_financial_report_webkit/models/account_static_balance.py

```diff
@@ -53,119 +61,155 @@ def unlink(self):
         closed periods and accounts if their combination is missing. """
-        periods, accounts = self.get_missing_periods_and_accounts()
+        periods, accounts, journals = \
+            self.get_missing_periods_accounts_and_journals()
         if not periods:
             logger.debug("No missing entries found. Static account balance "
                          "data is up to date.")
             return
-        self.calculate_static_balance(periods=periods, accounts=accounts)
+        self.calculate_static_balance(
+            periods=periods, accounts=accounts, journals=journals)
 
     @api.multi
-    def get_missing_periods_and_accounts(self, periods=None):
+    def get_missing_periods_accounts_and_journals(self, periods=None):
         # Split into unique periods and accounts
-        per_ids = acc_ids = []
+        per_ids = []
+        acc_ids = []
+        journal_ids = []
         missing_combinations = self.get_entries_to_calculate(periods=periods)
         if missing_combinations:
-            per_ids, acc_ids = map(list, map(set, zip(*missing_combinations)))
+            per_ids, acc_ids, journal_ids = map(
+                list, map(set, zip(*missing_combinations)))
         missing_periods = self.env['account.period'].browse(per_ids)
         missing_accounts = self.env['account.account'].browse(acc_ids)
-        return missing_periods, missing_accounts
+        missing_journals = self.env['account.journal'].browse(journal_ids)
+        return missing_periods, missing_accounts, missing_journals
 
     @api.model
-    def calculate_static_balance(self, periods, accounts=None):
+    def calculate_static_balance(self, periods, accounts=None, journals=None):
         """ Calculate the balance for a specified set of periods
         and insert them into the table."""
         self = self.suspend_security()
-        periods = periods.filtered(
-            lambda p: p.state == 'done' and not p.special)
-        for row in self.calculate_balance(periods, accounts=accounts):
-            try:
-                with self.env.cr.savepoint():
-                    self.create(row)
-            except psycopg2.IntegrityError:
-                # A balance already exists for this period and account
-                pass
+
+        if not journals:
+            # When journals are set, we can assume that this is a close for
+            # a journal period. Otherwise we need to filter the periods to
+            # check if they are in a done state
+            periods = periods.filtered(
+                lambda p: p.state == 'done' and not p.special)
+
+        for period in periods:
+            for row in self.calculate_balance(
+                    period, accounts=accounts, journals=journals):
+                try:
+                    with self.env.cr.savepoint():
+                        self.create(row)
+                except psycopg2.IntegrityError:
+                    # A balance already exists for this period and account
```

Comment:
Perhaps we can log the error message as a warning instead? Just for debugging purposes?